File: Data Structures & Algorithms/islands-and-treasure/submission-1.cpp
class Solution {
public:
    void islandsAndTreasure(vector<vector<int>>& grid) {

        queue<pair<int,int>> q;
        int m = grid.size();
        int n = grid[0].size();


        vector<vector<int>> visited_array(m, vector<int>(n, 0));

        for(int i = 0;i<m;i++){
            for(int j=0;j<n;j++){
                if(grid[i][j]==0){
                    q.push({i,j});
                    visited_array[i][j] = 1;
                }
            }
        }

        

        while (!q.empty()){
            pair<int,int> index = q.front();
            q.pop();
            
            int i = index.first;
            int j = index.second;
            

            if(i-1>=0 and visited_array[i-1][j] == 0 and grid[i-1][j]!= -1){
                grid[i-1][j] = grid[i][j] + 1;
                q.push({i-1,j});
                visited_array[i-1][j] = 1;
            }

            if(i+1<m and visited_array[i+1][j] == 0 and grid[i+1][j]!= -1){
                grid[i+1][j] = grid[i][j] + 1;
                q.push({i+1,j});
                visited_array[i+1][j] = 1;
            }
            if(j-1>=0 and visited_array[i][j-1] == 0 and grid[i][j-1]!= -1){
                grid[i][j-1] = grid[i][j] + 1;
                q.push({i,j-1});
                visited_array[i][j-1] = 1;
            }
            if(j+1<n and visited_array[i][j+1] == 0 and grid[i][j+1]!= -1){
                grid[i][j+1] = grid[i][j] + 1;
                q.push({i,j+1});
                visited_array[i][j+1] = 1;
            }



        }
        
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-1.py
class Solution:
    def countSubstrings(self, s: str) -> int:

        count = 0
        length = len(s)

        for i in range(length):
            count+=1
            start = i-1
            end = i+1
            while(start>=0 and end<length):
                if(s[start] == s[end]):
                    count+=1
                    start = start - 1
                    end = end+1
                
                else:
                    break
        

        for i in range(length-1):
            first = i
            second = i+1
            if(s[first] == s[second]):
                count+=1
            
                start = first-1
                end = second+1
                while(start>=0 and end<length):
                    if(s[start] ==s[end]):
                        count+=1
                        start = start - 1
                        end = end+1
                    
                    else:
                        break

        
        return count


        

        
        